### PR TITLE
refactor(dropdown): use colors from schemas

### LINKED ISSUE #828: Dropdown: should consume colors from schemas
## Result  
Currently, the drop-down is not consuming colors from the schemas.

## Expected result  
The drop-down should consume colors from the schemas.

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #828

### CODE REVIEW

File: src/components/dropdown/themes/light/dropdown.bootstrap.scss [hunk 1]
```diff
@@ -10,14 +10,14 @@ $theme: digest-schema(extend(
...
     )
 ));
-$border-color: var(--border-color, color(gray, 300)) !default;
 
 :host {
     @include css-vars-from-theme($theme, 'ig-dropdown');
 }
 
 [part='base'] {
+    background: color(surface, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/dropdown.bootstrap.scss:19]: Shouldn't this be consumed from the schema file?

File: src/components/dropdown/themes/light/dropdown.bootstrap.scss [hunk 2]
```diff
@@ -10,14 +10,14 @@ $theme: digest-schema(extend(
...
     border-radius: var-get($theme, 'border-radius');
-    box-shadow: none;
-    border: 1px solid $border-color;
+    padding: rem(8px) 0;
+    border: rem(1px) solid var(--border-color, color(gray, 300));
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/dropdown.bootstrap.scss:22]: How about the border-color here?

File: src/components/dropdown/themes/light/dropdown.fluent.scss [hunk 3]
```diff
@@ -16,7 +16,7 @@ $theme: digest-schema(extend(
 }
 
 [part='base'] {
+    background: color(surface, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/dropdown.fluent.scss:19]: Schema?

File: src/components/dropdown/themes/light/dropdown.indigo.scss [hunk 4]
```diff
@@ -16,7 +16,7 @@ $theme: digest-schema(extend(
 }
 
 [part='base'] {
+    background: color(surface, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/dropdown.indigo.scss:19]: Schema?

File: src/components/dropdown/themes/light/dropdown.material.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+}
+
+[part='base'] {
+    background: color(surface, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/dropdown.material.scss:19]: Schema?

File: src/components/dropdown/themes/light/group/dropdown-group.bootstrap.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    ::slotted([slot='label']) {
+        color: color(secondary, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.bootstrap.scss:18]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/group/dropdown-group.fluent.scss [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    ::slotted([slot='label']) {
+        color: color(secondary, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.fluent.scss:18]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/group/dropdown-group.fluent.scss [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        padding: pad(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.fluent.scss:19]:
```suggestion
        padding: pad(rem(2px), rem(4px), (8px));
```

File: src/components/dropdown/themes/light/group/dropdown-group.indigo.scss [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    ::slotted([slot='label']) {
+        color: color(secondary, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.indigo.scss:18]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/group/dropdown-group.material.scss [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    ::slotted([slot='label']) {
+        color: color(secondary, 500);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.material.scss:18]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/header/dropdown-header.bootstrap.scss [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    $schema,
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    color: color(gray, 700);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.bootstrap.scss:17]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/header/dropdown-header.bootstrap.scss [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    padding-inline: pad-inline(4px, 8px, 16px);
+    padding-block: pad-block(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.bootstrap.scss:19]:
```suggestion
    padding-inline: pad-inline(rem(4px), rem(8px), rem(16px));
    padding-block: pad-block(rem(2px), rem(4px), rem(8px));
```

File: src/components/dropdown/themes/light/header/dropdown-header.fluent.scss [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    $schema,
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    color: var(--header-color, color(secondary, 500));
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.fluent.scss:17]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/header/dropdown-header.fluent.scss [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+    padding: pad(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.fluent.scss:18]:
```suggestion
    padding: pad(rem(2px), rem(4px), rem(8px));
```

File: src/components/dropdown/themes/light/header/dropdown-header.indigo.scss [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+    $schema,
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    color: var(--header-color, color(secondary, 500));
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.indigo.scss:17]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/header/dropdown-header.indigo.scss [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+    padding-inline: pad-inline(4px, 8px, 16px);
+    padding-block: pad-block(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.indigo.scss:19]:
```suggestion
    padding-inline: pad-inline(rem(4px), rem(8px), rem(16px));
    padding-block: pad-block(rem(2px), rem(4px), rem(8px));
```

File: src/components/dropdown/themes/light/header/dropdown-header.material.scss [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+    $schema,
+    (
+        border-radius: $border-radius
+    )
+));
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    color: var(--header-color, color(secondary, 500));
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.material.scss:17]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/header/dropdown-header.material.scss [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+    padding-inline: pad-inline(4px, 8px, 16px);
+    padding-block: pad-block(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/header/dropdown-header.material.scss:19]:
```suggestion
    padding-inline: pad-inline(rem(4px), rem(8px), rem(16px));
    padding-block: pad-block(rem(2px), rem(4px), rem(8px));
```

File: src/components/dropdown/themes/light/item/dropdown-item.base.scss [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+    display: flex;
+    justify-content: flex-start;
+    align-items: center;
+    width: 100%;
+    white-space: nowrap;
+    position: relative;
+    cursor: pointer;
+    outline-style: none;
+    padding-inline: pad-inline(6px, 12px, 24px);
+    padding-block: pad-block(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/item/dropdown-item.base.scss:16]:
```suggestion
    padding-inline: pad-inline(rem(6px), rem(12px), rem(24px));
    padding-block: pad-block(rem(2px), rem(4px), rem(8px));
```

File: src/components/dropdown/themes/light/item/dropdown-item.base.scss [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+
+[part='prefix'],
+[part='suffix'] {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+[name='prefix']::slotted(*) {
+    margin-inline-end: pad(16px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/item/dropdown-item.base.scss:44]:
```suggestion
    margin-inline-end: pad(rem(16px));
```

File: src/components/dropdown/themes/light/item/dropdown-item.base.scss [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+[name='suffix']::slotted(*) {
+    margin-inline-start: pad(16px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/item/dropdown-item.base.scss:48]:
```suggestion
    margin-inline-start: pad(rem(16px));
```

File: src/components/dropdown/themes/light/item/dropdown-item.bootstrap.scss [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+
+$schema: map.get($light-bootstrap-schema, 'drop-down');
+$item-border-radius: border-radius(map.get($schema, 'item-border-radius')...);
+$theme: digest-schema(extend(
+    $schema,
+    (
+        item-border-radius: $item-border-radius
+    )
+));
+$hover-item-color: var(--hover-item-color, contrast-color(gray, 100)) !default;
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/item/dropdown-item.bootstrap.scss:13]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/item/dropdown-item.fluent.scss [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+
+$schema: map.get($light-fluent-schema, 'drop-down');
+$item-border-radius: border-radius(map.get($schema, 'item-border-radius')...);
+$theme: digest-schema(extend(
+    $schema,
+    (
+        item-border-radius: $item-border-radius
+    )
+));
+$selected-item-color: var(--selected-item-color, contrast-color(gray, 200)) !default;
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/item/dropdown-item.fluent.scss:13]: Shouldn't this come from the schema?

File: src/components/dropdown/themes/light/item/dropdown-item.fluent.scss [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+
+:host {
+    @include css-vars-from-theme($theme, 'ig-dropdown');
+
+    border-radius: var-get($theme, 'item-border-radius');
+    color: var-get($theme, 'item-text-color');
+    padding: pad(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/item/dropdown-item.fluent.scss:20]:
```suggestion
    padding: pad(rem(2px), rem(4px), rem(8px));
```

File: src/components/dropdown/themes/light/group/dropdown-group.base.scss [hunk 25]
```diff
@@ -11,7 +11,6 @@
         display: block;
         white-space: nowrap;
         position: relative;
-        color: color(secondary, 500);
         pointer-events: none;
         padding-inline: pad-inline(4px, 8px, 16px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.base.scss:15]:
```suggestion
        padding-inline: pad-inline(rem(4px), rem(8px), rem(16px));
```

File: src/components/dropdown/themes/light/group/dropdown-group.base.scss [hunk 26]
```diff
@@ -11,7 +11,6 @@
...
         padding-block: pad-block(2px, 4px, 8px);
```
Comment (simeonoff) [inline src/components/dropdown/themes/light/group/dropdown-group.base.scss:16]:
```suggestion
        padding-block: pad-block(rem(2px), rem(4px), rem(8px));
```

**Review by simeonoff — CHANGES_REQUESTED** [review]

**Comment by desig9stein** [comment]
@simeonoff I need to implement the dark themes as well

**Review by simeonoff — APPROVED** [review]

**Review by simeonoff — APPROVED** [review]

**Pull request merged**